python/common/org/python/types/Str.java

```diff
@@ -1190,10 +1190,46 @@ private long toPositiveIndex(long index) {
                     "\n" +
                     "Search for the separator sep in S, starting at the end of S, and return\n" +
                     "the part before it, the separator itself, and the part after it.  If the\n" +
-                    "separator is not found, return two empty strings and S.\n"
+                    "separator is not found, return two empty strings and S.\n",
+            args = {"sep"}
     )
-    public org.python.Object rpartition() {
-        throw new org.python.exceptions.NotImplementedError("rpartition() has not been implemented.");
+    public org.python.Object rpartition(org.python.types.Object sep) {
+        java.lang.String sepStr = ((org.python.types.Str) sep).value;
+        java.lang.String str_one = new java.lang.String();
+        java.util.List<org.python.Object> tuple = new java.util.ArrayList<org.python.Object>();
+        if (sepStr.equals("")) {
+            throw new org.python.exceptions.ValueError("empty separator");
+        }
+        java.lang.String[] split = this.value.split(sepStr);
+        if (this.value.length() < sepStr.length()) {
+            tuple.add(new org.python.types.Str(""));
+            tuple.add(new org.python.types.Str(""));
+            tuple.add(new org.python.types.Str(this.value));
+            return new org.python.types.Tuple(tuple);
+        }
+        if (split.length == 1) {
+            if ((this.value.substring(this.value.length() - sepStr.length())).equals(sepStr)) {
```

Comment:
This condition seems a bit strange. In which case will it be true? It seems like you're taking the string of the last `sepStr.length()` chars and comparing it with `sepStr` itself?